lib/_http_client.js

```diff
@@ -619,6 +612,17 @@ function parserOnIncomingClient(res, shouldKeepAlive) {
 // client
 function responseKeepAlive(req) {
   const socket = req.socket;
+  // There are cases where _handle === null. Avoid those. Passing null to
+  // nextTick() will call getDefaultTriggerAsyncId() to retrieve the id.
+  const asyncId = socket._handle ? socket._handle.getAsyncId() : undefined;
+  // Mark this socket as available, AFTER user-added end
+  // handlers have a chance to run.
+  defaultTriggerAsyncIdScope(asyncId, process.nextTick, responseKeepAliveNT,
```

Comment:
This changes when the close/error/data/end listeners are removed (one tick later now.)

I _think_ that's okay but again, it's hard to reason about.

Reply:
Correct, the problem is that as it was before the `'error'` handler was removed too early, causing potential unhandled exception.